src/login/logind-dbus.c

```diff
@@ -2382,6 +2383,80 @@ static int method_can_suspend_then_hibernate(sd_bus_message *message, void *user
                         error);
 }
 
+static int property_get_reboot_argument(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+        _cleanup_free_ char *parameter = NULL;
+        int r;
+
+        assert(bus);
+        assert(reply);
+        assert(userdata);
+
+        r = get_reboot_parameter(&parameter);
+        if (r != 0)
+                return r;
+
+        return sd_bus_message_append(reply, "s", parameter);
+}
+
+static int method_set_reboot_argument(
+                sd_bus_message *message,
+                void *userdata,
+                sd_bus_error *error) {
+
+        Manager *m = userdata;
+        const char *arg;
+        int r;
+
+        assert(message);
+        assert(m);
+
+        r = sd_bus_message_read(message, "s", &arg);
+        if (r < 0)
+                return r;
+
+        r = verify_shutdown_creds(
+                        m,
+                        message,
+                        INHIBIT_SHUTDOWN,
+                        false,
+                        "org.freedesktop.login1.reboot",
+                        "org.freedesktop.login1.reboot-multiple-sessions",
+                        "org.freedesktop.login1.reboot-ignore-inhibit",
+                        error);
+        if (r != 0)
+                return r;
+
+        r = update_reboot_parameter_and_warn(arg, false);
+        if (r != 0)
+                return r;
+
+        return sd_bus_reply_method_return(message, NULL);
+}
+
+static int method_can_reboot_argument(
+                sd_bus_message *message,
+                void *userdata,
+                sd_bus_error *error) {
+
+        Manager *m = userdata;
+
+        return method_can_shutdown_or_sleep(
+                        m, message,
+                        INHIBIT_SHUTDOWN,
+                        "org.freedesktop.login1.reboot",
+                        "org.freedesktop.login1.reboot-multiple-sessions",
+                        "org.freedesktop.login1.reboot-ignore-inhibit",
+                        NULL,
+                        error);
+}
```

Comment:
this doesn't look right. The check you are doing here is for the action of rebooting, not the ability to set the boot parameter.

This should be modelled after method_can_reboot_to_firmware_setup() or method_can_reboot_to_boot_loader_menu(). It should probably hand off quickly to return_test_polkit(), but say that the feature is not supported in containers (as the kernel doesn't do anything with the reboot param in that case, there's no way to propagate it).

i.e.:

```c
static int method_can_reboot_param(
                sd_bus_message *message,
                void *userdata,
                sd_bus_error *error) {

        Manager *m = userdata;
        int r;

        assert(message);
        assert(m);

        r = detect_container();
        if (r < 0)
                return r;
        if (r > 0) /* Inside containers, specifiying a reboot parameter, doesn't make much sense */
                 return sd_bus_reply_method_return(message, "s", "na");

        return return_test_polkit(
                        message,
                        CAP_SYS_ADMIN,
                        "org.freedesktop.login1.set-reboot-param",
                        NULL,
                        UID_INVALID,
                        error);
}
```

Reply:
Ok. Changed as suggested. Also renamed the RebootArgument as RebootParam as suggested with the naming example in here.